.github/workflows/performance_tests.yml

```diff
@@ -0,0 +1,53 @@
+name: Performance Tests
+
+on:
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron:  '0 9 * * 0'
+
+jobs:
+  check_branch_changes:
+    runs-on: ubuntu-22.04
+    outputs:
+      changed: ${{ steps.branch_change_output.outputs.has-new-commits }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: check branch changes
+        id: branch_change_output
+        uses: adriangl/check-new-commits-action@v1
+        with:
+          seconds: 604800
+          branch: 'dev'
+
+  run_performance_tests:
+    runs-on: ubuntu-22.04
+    needs: check_branch_changes
+    if: ${{ needs.check_branch_changes.outputs.changed == 'true' }}
+    defaults:
+      run:
+        working-directory: ./test/performance
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: 'main'
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.1'
+      - run: bundle
+      - run: script/runner -B
+      - uses: actions/checkout@v3
+      - run: bundle
+      - run: script/runner -C > performance_tests.txt
+      - name: Save performance results
+        uses: actions/upload-artifact@v3
+        with:
+          name: performance-test-results
+          path: test/performance/performance_tests.txt
+      - name: Slack results
+        uses: adrey/slack-file-upload-action@1.0.5
+        with:
+          token: ${{ secrets.RUBY_GITHUB_ACTIONS_BOT_WEBHOOK }}
+          path: ./test/performance/performance_tests.txt
```

Comment:
Nitpicking things for possible future PR fixes...

- There's a `working-directory` set, but then the `path` values repeat the `test/performance` path. Can they use `performance_tests.txt` by itself, or is the working directory not applicable to the paths?
- The first `path` uses `test` and the second `path` uses `./test`. Both are fine, but they ought to match.
- Given that the descriptions all say "results", perhaps the filename should be `performance_results.txt`.
- Once the `runner` script arguments are updated in future, the filename can use `md` for the extension

Reply:
I love your detailed feedback. Thank you; I'll save these notes in a ticket for the future.